ArjunaJTA/jdbc/classes/com/arjuna/ats/internal/jdbc/recovery/BasicXARecovery.java

```diff
@@ -157,7 +158,10 @@ public synchronized XAResource getXAResource () throws SQLException {
 			connections = new JDBC2RecoveryConnection[numberOfConnections];
 		}
 		try {
-			return getConnection().recoveryConnection().getResource();
+		    RecoverableXAConnection recoveryConnection = getConnection().recoveryConnection();
+		    if(recoveryConnection == null)
+		        throw new IllegalStateException("Recovery connection of connection " + getConnection() + " is null");
```

Comment:
I assume the XARM will be able to deal with this OK?

Reply:
@tomjenkinson yes, I assume so. The XARM catches any exception and logs it (https://github.com/jbosstm/narayana/blob/master/ArjunaJTA/jta/classes/com/arjuna/ats/internal/jta/recovery/arjunacore/XARecoveryModule.java#L551) (e.g. `#getXAResource` throws the SQLE). I think let the NPE being thrown non-intentionally should be avoided thus I changed it to explicitly stating the state is illegal/unexpected. But maybe the NPE could be left here.